machineconfiguration/v1alpha1/types_internalreleaseimage.go

```diff
@@ -0,0 +1,145 @@
+	// +required
+	Name string `json:"name,omitempty"`
+	// image is an OCP release image referenced by digest.
+	// The format of the image pull spec is: host[:port][/namespace]/name@sha256:<digest>,
+	// where the digest must be 64 characters long, and consist only of lowercase hexadecimal characters, a-f and 0-9.
+	// The length of the whole spec must be between 0 to 447 characters.
+	// The field is optional, and it will be provided after a release will be successfully installed.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=447
+	// +kubebuilder:validation:XValidation:rule=`self == '' || (self.split('@').size() == 2 && self.split('@')[1].matches('^sha256:[a-f0-9]{64}$'))`,message="the OCI Image reference must end with a valid '@sha256:<digest>' suffix, where '<digest>' is 64 characters long"
+	// +kubebuilder:validation:XValidation:rule=`self == '' || (self.split('@')[0].matches('^([a-zA-Z0-9-]+\\.)+[a-zA-Z0-9-]+(:[0-9]{2,5})?/([a-zA-Z0-9-_]{0,61}/)?[a-zA-Z0-9-_.]*?$'))`,message="the OCI Image name should follow the host[:port][/namespace]/name format, resembling a valid URL without the scheme"
+	// +optional
+	Image string `json:"image,omitempty"`
+}
+
+// InternalReleaseImageConditionType is each possible state for each possible MachineConfigNodeStatusInternalReleaseImageRef
+// conditions type.
+// +enum
+type InternalReleaseImageConditionType string
+
+const (
+	// InternalReleaseImageConditionTypeMounted describes a new release, not yet installed, that has been discovered when an ISO has been attached to
+	// one of the control plane nodes
+	InternalReleaseImageConditionTypeMounted InternalReleaseImageConditionType = "Mounted"
+	// InternalReleaseImageConditionTypeInstalling describes a new release that is getting installed in the cluster. Due the size of the data
+	// transfered, the operation could take several minutes. The condition will remain in such state until all the control plane nodes will
+	// complete the installing operation
+	InternalReleaseImageConditionTypeInstalling InternalReleaseImageConditionType = "Installing"
+	// InternalReleaseImageConditionTypeAvailable describes a release that has been successfully installed in the cluster, ready to be consumed. This
+	// means that the release has been successfully installed on all the control plane nodes
+	InternalReleaseImageConditionTypeAvailable InternalReleaseImageConditionType = "Available"
+	// InternalReleaseImageConditionTypeRemoving describes an existing release that is getting removed from the cluster. The condition will remain in such
+	// state until all the control plane nodes will complete the removal operation
+	InternalReleaseImageConditionTypeRemoving InternalReleaseImageConditionType = "Removing"
+	// InternalReleaseImageConditionTypeDegraded describes a failure, happened in one or more control plane nodes, for the current release
+	InternalReleaseImageConditionTypeDegraded InternalReleaseImageConditionType = "Degraded"
+)
```

Comment:
So in a steady state, I'd expect to see all five of these conditions, in order with status False, False, True, False, False, right?